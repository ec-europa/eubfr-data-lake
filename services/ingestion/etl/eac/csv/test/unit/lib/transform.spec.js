/**
 * @jest-environment node
 */

// There are many files in this ETL, and for this reason variables are same as file names for easier discovery.
/* eslint camelcase: 0 */

import mapper from '../../../src/lib/transform';
import testCoopIndustrialisedCountries_Projects_Overview from '../../stubs/CoopIndustrialisedCountries_Projects_Overview.json';
import testCreativeEurope_Projects_Overview from '../../stubs/CreativeEurope_Projects_Overview.json';
import testCulture_2007_2013_Projects_Overview from '../../stubs/Culture_2007_2013_Projects_Overview.json';
import testErasmusMundus_Projects_Overview from '../../stubs/ErasmusMundus_Projects_Overview.json';
import testErasmusPlus_JeanMonnet_Projects_Overview from '../../stubs/ErasmusPlus_JeanMonnet_Projects_Overview.json';
import testErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview from '../../stubs/ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview.json';
import testErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview from '../../stubs/ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview.json';
import testErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview from '../../stubs/ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview.json';
import testErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview from '../../stubs/ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview.json';
import testErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview from '../../stubs/ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview.json';
import testErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview from '../../stubs/ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview.json';
import testErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview from '../../stubs/ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview.json';
import testErasmusPlus_Sports_Projects_Overview from '../../stubs/ErasmusPlus_Sports_Projects_Overview.json';
import testLifeLongLearning_Projects_Overview from '../../stubs/LifeLongLearning_Projects_Overview.json';
import testSports_Projects_Overview from '../../stubs/Sports_Projects_Overview.json';
import testTempus_Projects_Overview from '../../stubs/Tempus_Projects_Overview.json';
import testYouthInAction_Projects_Overview from '../../stubs/YouthInAction_Projects_Overview.json';

describe('DG EAC CSV transformer', () => {
  let CoopIndustrialisedCountries_Projects_Overview = {};
  let CreativeEurope_Projects_Overview = {};
  let Culture_2007_2013_Projects_Overview = {};
  let ErasmusMundus_Projects_Overview = {};
  let ErasmusPlus_JeanMonnet_Projects_Overview = {};
  let ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview = {};
  let ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview = {};
  let ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview = {};
  let ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview = {};
  let ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview = {};
  let ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview = {};
  let ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview = {};
  let ErasmusPlus_Sports_Projects_Overview = {};
  let LifeLongLearning_Projects_Overview = {};
  let Sports_Projects_Overview = {};
  let Tempus_Projects_Overview = {};
  let YouthInAction_Projects_Overview = {};

  beforeAll(() => {
    CoopIndustrialisedCountries_Projects_Overview = mapper(
      testCoopIndustrialisedCountries_Projects_Overview
    );
    CreativeEurope_Projects_Overview = mapper(
      testCreativeEurope_Projects_Overview
    );
    Culture_2007_2013_Projects_Overview = mapper(
      testCulture_2007_2013_Projects_Overview
    );
    ErasmusMundus_Projects_Overview = mapper(
      testErasmusMundus_Projects_Overview
    );
    ErasmusPlus_JeanMonnet_Projects_Overview = mapper(
      testErasmusPlus_JeanMonnet_Projects_Overview
    );
    ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview = mapper(
      testErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview
    );
    ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview = mapper(
      testErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview
    );
    ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview = mapper(
      testErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview
    );
    ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview = mapper(
      testErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview
    );
    ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview = mapper(
      testErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview
    );
    ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview = mapper(
      testErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview
    );
    ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview = mapper(
      testErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview
    );
    ErasmusPlus_Sports_Projects_Overview = mapper(
      testErasmusPlus_Sports_Projects_Overview
    );
    LifeLongLearning_Projects_Overview = mapper(
      testLifeLongLearning_Projects_Overview
    );
    Sports_Projects_Overview = mapper(testSports_Projects_Overview);
    Tempus_Projects_Overview = mapper(testTempus_Projects_Overview);
    YouthInAction_Projects_Overview = mapper(
      testYouthInAction_Projects_Overview
    );
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure for CoopIndustrialisedCountries_Projects_Overview', () => {
    expect(CoopIndustrialisedCountries_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for CoopIndustrialisedCountries_Projects_Overview do not contain decimals', () => {
    expect(
      CoopIndustrialisedCountries_Projects_Overview.budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      CoopIndustrialisedCountries_Projects_Overview.budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for CreativeEurope_Projects_Overview', () => {
    expect(CreativeEurope_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for CreativeEurope_Projects_Overview do not contain decimals', () => {
    expect(CreativeEurope_Projects_Overview.budget.eu_contrib.value % 1).toBe(
      0
    );
    expect(CreativeEurope_Projects_Overview.budget.total_cost.value % 1).toBe(
      0
    );
  });

  test('Produces correct JSON output structure for Culture_2007_2013_Projects_Overview', () => {
    expect(Culture_2007_2013_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for Culture_2007_2013_Projects_Overview do not contain decimals', () => {
    expect(
      Culture_2007_2013_Projects_Overview.budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      Culture_2007_2013_Projects_Overview.budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusMundus_Projects_Overview', () => {
    expect(ErasmusMundus_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for ErasmusMundus_Projects_Overview do not contain decimals', () => {
    expect(ErasmusMundus_Projects_Overview.budget.eu_contrib.value % 1).toBe(0);
    expect(ErasmusMundus_Projects_Overview.budget.total_cost.value % 1).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_JeanMonnet_Projects_Overview', () => {
    expect(ErasmusPlus_JeanMonnet_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_JeanMonnet_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_JeanMonnet_Projects_Overview.budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_JeanMonnet_Projects_Overview.budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA1_2014_LearningMobilityOfIndividuals_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA1_2015_LearningMobilityOfIndividuals_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA1_2016_LearningMobilityOfIndividuals_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA1_2017_LearningMobilityOfIndividuals_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA1_2018_LearningMobilityOfIndividuals_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview
        .budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA2_CooperationForInnovationAndTheExchangeOfGoodPractices_Projects_Overview
        .budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview', () => {
    expect(
      ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview
    ).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview.budget.eu_contrib
        .value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_KA3_SupportForPolicyReform_Projects_Overview.budget.total_cost
        .value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for ErasmusPlus_Sports_Projects_Overview', () => {
    expect(ErasmusPlus_Sports_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for ErasmusPlus_Sports_Projects_Overview do not contain decimals', () => {
    expect(
      ErasmusPlus_Sports_Projects_Overview.budget.eu_contrib.value % 1
    ).toBe(0);
    expect(
      ErasmusPlus_Sports_Projects_Overview.budget.total_cost.value % 1
    ).toBe(0);
  });

  test('Produces correct JSON output structure for LifeLongLearning_Projects_Overview', () => {
    expect(LifeLongLearning_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for LifeLongLearning_Projects_Overview do not contain decimals', () => {
    expect(LifeLongLearning_Projects_Overview.budget.eu_contrib.value % 1).toBe(
      0
    );
    expect(LifeLongLearning_Projects_Overview.budget.total_cost.value % 1).toBe(
      0
    );
  });

  test('Produces correct JSON output structure for Sports_Projects_Overview', () => {
    expect(Sports_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for Sports_Projects_Overview do not contain decimals', () => {
    expect(Sports_Projects_Overview.budget.eu_contrib.value % 1).toBe(0);
    expect(Sports_Projects_Overview.budget.total_cost.value % 1).toBe(0);
  });

  test('Produces correct JSON output structure for Tempus_Projects_Overview', () => {
    expect(Tempus_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for Tempus_Projects_Overview do not contain decimals', () => {
    expect(Tempus_Projects_Overview.budget.eu_contrib.value % 1).toBe(0);
    expect(Tempus_Projects_Overview.budget.total_cost.value % 1).toBe(0);
  });

  test('Produces correct JSON output structure for YouthInAction_Projects_Overview', () => {
    expect(YouthInAction_Projects_Overview).toMatchSnapshot();
  });

  test('Budget values for YouthInAction_Projects_Overview do not contain decimals', () => {
    expect(YouthInAction_Projects_Overview.budget.eu_contrib.value % 1).toBe(0);
    expect(YouthInAction_Projects_Overview.budget.total_cost.value % 1).toBe(0);
  });
});
