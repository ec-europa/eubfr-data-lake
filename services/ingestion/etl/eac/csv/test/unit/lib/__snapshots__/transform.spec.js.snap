// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG EAC CSV transformer Produces correct JSON output structure for CoopIndustrialisedCountries_Projects_Overview 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
  },
  "call_year": "",
  "complete": true,
  "description": "Using the successful model of airline alliances, the EU/Korea Leadership Alliance (EUKLA) will establish a framework
within which international activities between the EU and Korea will be promoted, recorded, recognized, and rewarded.
With involvement from key industries with vested links in the EU and Korea, EUKLA will establish clear point-based
pathways to engage engineering and business students in international activities leading to recognized outcomes with
real career benefits. A total of 84 EU students and 60 Korean students will participate in the mobility component to
accumulate overseas activity points, covering intercultural activities, foreign language acquisition, industrial placements,
and research activities. Completion of a pathway will then be acknowledged with a special EUKLA award status: silver,
gold, or platinum, jointly recognized by all the university, industry, embassy, and local government partners. As a result,
EUKLA will build a strong synergy between the partners to promote mobility, recognize achievement, and meet the
needs of EU/Korean industry, thereby securing the long-term growth and success of EU/Korean relations.",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "Education and training\\\\External Programmes and Policies\\\\Co-operation with industrialised countries\\\\Co-operation with other industrialised countries",
  "project_id": "EACEA-168191",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "PL",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HU",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "LT",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "DE",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "KR",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "EAC",
  "results": Object {
    "available": "No",
    "result": "https://ec.europa.eu/programmes/erasmus-plus/projects/eplus-project-details/#project/168191-EM-1-2009-1-PL-ERA MUNDUS-ICI",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "No",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "Pl. Politechniki, 1,  00661,  WARSZAW",
      "country": "PL",
      "email": "",
      "name": "POLITECHNIKA WARSZAWSKA",
      "phone": "",
      "region": "",
      "role": "coordinator",
      "type": "",
      "website": "",
    },
    Object {
      "address": "Müegyetem rkp 3-9  1111 BUDAPEST",
      "country": "HU",
      "email": "",
      "name": "BUDAPEST UNIVERSITY OF TECHNOLOGY AND ECONOMICS",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "Saulétekio al.11  10223 VILNIUS",
      "country": "LT",
      "email": "",
      "name": "VILNIUS GEDIMINAS TECHNICAL UNIVERSITY",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "Max-Planck Str. 39  74081 HEILBRONN",
      "country": "DE",
      "email": "",
      "name": "UNIVERSITY OF APPLIED SCIENCES HEILBRONN",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "Daehak-ro 102, Nam-gu   ULSAN",
      "country": "KR",
      "email": "",
      "name": "UNIVERSITY OF ULSAN",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "1370 Sankyuk-dong, Buk-gu  702-701 DAEGU",
      "country": "KR",
      "email": "",
      "name": "KYUNGPOOK NATIONAL UNIVERSITY",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
    Object {
      "address": "1 Yangho-dong Gumi Gyeongbuk-do  730-701 GYEONGBUK-DO",
      "country": "KR",
      "email": "",
      "name": "KUMOH NATIONAL INSTITUTE OF TECHNOLOGY",
      "phone": "",
      "region": "",
      "role": "partner",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": null,
    "from_precision": "day",
    "to": null,
    "to_precision": "day",
  },
  "title": "KR - EUKLA - European Korean Leadership Alliance",
  "type": Array [],
}
`;

exports[`DG EAC CSV transformer Produces correct JSON output structure for CreativeEurope_Projects_Overview 1`] = `
Object {
  "action": "Cooperation projects",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": "200000.00",
      "value": 200000,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
  },
  "call_year": "2014",
  "complete": true,
  "description": "The ELiT Literaturehouse Europe establishes an observatory for European contemporary literature focusing chiefly on: research, discussion and publishing results concerning literary trends across Europe, as well as the inter-cultural communication of literature within Europe and the dissemination of literature among the diverse cultural spaces within Europe. It enables writers to introduce their works in other linguistic areas and to discover new opportunities for publishing based on digital media. It develops innovative forms of communicating literature for young people from different south eastern European countries of the Danube Region Strategy who get to know Europe’s shared heritage through working on a literary subject and producing a collectively authored ebook.ELiT Literaturehouse Europe regularly organizes these events: European literature youth meetings, European writers’ reading tours sessions, in particular, with the winners of the European Union Prize for Literature, workshops for young writers and – the annual highlight – the European Literature Days in the Wachau/Lower Austria.ELiT Literaturehouse Europe headquarters is in Wachau/Lower Austria, where it facilitates networking for literary experts, centres and events as well as setting up cooperative programmes within Europe. It upgrades the professionalitiy of it‘s work and  currently sets up a programme for writers and translators-in-residence. The European Literature Days take place in Spitz, a little village in Wachau, an outstanding cultural landscape and UNESCO World Heritage Centre. They are a highlight of the annual culture programme of Spitz and both well known and estimated in the region. At the website www.literaturehouse.eu,  the European Literature social media platform www.readme.cc and trough ebook- and book on demand publications ELiT Literaturehouse Europe will introduce literary trends in Europe and publish and discuss texts on European issues in cultural policy.",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "Creative Europe",
  "project_id": "552144-CREA-1-2014-1-AT-CULT-COOP1",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "AT",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "GB",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "DE",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "FR",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HU",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
    Object {
      "address": "",
      "centroid": null,
      "country_code": "SI",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "EAC",
  "results": Object {
    "available": "No",
    "result": "https://ec.europa.eu/programmes/creative-europe/projects/ce-project-details/#project/552144-CREA-1-2014-1-AT-CULT-COOP1",
  },
  "status": "ongoing",
  "sub_programme_name": "Culture",
  "success_story": "No",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "JACOB PRANDTAUER STRASSE 11, 3390, MELK",
      "country": "AT",
      "email": "",
      "name": "WACHAU KULTUR MELK GMBH",
      "phone": "",
      "region": "Niederösterreich",
      "role": "coordinator",
      "type": "Multimedia association",
      "website": "www.kultur-melk.at",
    },
    Object {
      "address": "94 GOLDSMITH AVENUE, W3 6HW, LONDON",
      "country": "UK",
      "email": "",
      "name": "EASTERMAN MAX",
      "phone": "",
      "region": "Inner London",
      "role": "partner",
      "type": "Other",
      "website": "www.soundsright.biz",
    },
    Object {
      "address": "SCHWANENWIK 38, 22087, HAMBURG",
      "country": "DE",
      "email": "",
      "name": "LITERATURHAUS EV",
      "phone": "",
      "region": "HAMBURG",
      "role": "partner",
      "type": "Literature Foundation",
      "website": "www.literaturhaus-hamburg.de",
    },
    Object {
      "address": "RUE DE LA SORBONNE 17, 75005, PARIS",
      "country": "FR",
      "email": "",
      "name": "UNIVERSITE PARIS III SORBONNE NOUVELLE",
      "phone": "",
      "region": "Extra-Regio NUTS 2",
      "role": "partner",
      "type": "Higher education institution (tertiary level)",
      "website": "www.univ-paris3.fr",
    },
    Object {
      "address": "PAULAY EDE UTCA 19 1/1, 1061, BUDAPEST",
      "country": "HU",
      "email": "",
      "name": "EUROPAI KULTURALIS ALAPITVANY BUDAPEST",
      "phone": "",
      "region": "Budapest",
      "role": "partner",
      "type": "Multimedia association",
      "website": "ecfbudapest.com",
    },
    Object {
      "address": "MINORITENPLATZ 4, 3500, KREMS AN DER DONAU",
      "country": "AT",
      "email": "",
      "name": "NO FESTIVAL UND KINO GMBH",
      "phone": "",
      "region": "Niederösterreich",
      "role": "partner",
      "type": "Cultural operators",
      "website": "www.literaturehouseeurope.eu",
    },
    Object {
      "address": "KERSNIKOVA ULICA 4, 1000, LJUBLJANA",
      "country": "SI",
      "email": "",
      "name": "BELETRINA, ZAVOD ZA ZALOZNISKO DEJAVNOST",
      "phone": "",
      "region": "Osrednjeslovenska",
      "role": "partner",
      "type": "",
      "website": "www.beletrina.si",
    },
  ],
  "timeframe": Object {
    "from": "2015-01-01T00:00:00.000Z",
    "from_precision": "day",
    "to": "2018-12-31T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "ELiT Literaturehouse Europe",
  "type": Array [
    "Creative writing",
    "Translation",
    "Publishing",
  ],
}
`;
